### PR TITLE
Hooks support

### PR DESCRIPTION
**Pull request by zxan1285**
## ✏️ Changes
  Added hooks support. Also contains changes from #43 
  
## 🎯 Testing
🚫 This change has been tested in a Webtask
✅ This change has unit test coverage
🚫 This change has integration test coverage
🚫 This change has been tested for performance
  
## 🚀 Deployment
🚫 This can be deployed any time
⚠️ `source-control-extension-tools` has to be released and added to webtask modules.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,7 +86,7 @@
     "auth0-extension-tools": "1.4.0",
     "auth0-extension-ui": "^1.0.1",
     "auth0-oauth2-express": "^1.1.8",
-    "auth0-source-control-extension-tools": "3.5.1",
+    "auth0-source-control-extension-tools": "dev",
```
Comment (zxan1285) [inline package.json]: Don't forget to update this version after releasing `auth0-source-control-extension-tools` with hooks support. And also change version in `externals` above.

**Comment by aagavin** [comment]
This is awesome. Is there an ETA on when this will be propagated to the bitbucket and gitlab extensions?

**Comment by shawnmclean** [comment]
@aagavin this is now on our iteration to complete in 2 weeks time.

**Comment by aagavin** [comment]
Awesome, thanks for the update @shawnmclean

File: tests/server/mocks/expected-data.js [hunk 2]
```diff
@@ -15,6 +15,24 @@ module.exports = {
       stage: 'login_success'
     }
   ],
+  hooks: [
+    {
+      script: 'hook1 script',
+      name: 'Hook-1',
+      triggerId: 'client-credentials',
+      enabled: true,
+      secrets: {},
```
Comment (shushen) [inline tests/server/mocks/expected-data.js:24]: Is there a reason `secrets` and `dependecies` are always dropped?
Would this lead to the secrets and dependencies being removed?

**Review by shushen — CHANGES_REQUESTED** [review]
Blocking PR for clarification on the processing of `secrets` and `dependencies` fields.

File: tests/server/mocks/expected-data.js (hunk 2 above)
Comment (shawnmclean) [reply tests/server/mocks/expected-data.js:24]: Looking at the test, this is the expected data based on the data we pass in: https://github.com/auth0-extensions/auth0-deploy-extensions/pull/44/files#diff-298ad2b70dc85b087647efb12f2e14b8R85

The sourcecontrol-extension-tools library is what will handle the diffing and determine if  it needs to be updated, removed, etc.

Comment (shushen) [reply tests/server/mocks/expected-data.js:24]: Awesome, thanks for the clarifications!

**Review by shushen — APPROVED** [review]

**Review by shushen — CHANGES_REQUESTED** [review]
Sorry, one more thing - could you please update the [changelog](https://github.com/auth0-extensions/auth0-deploy-extensions/blob/master/CHANGELOG.md)?

**Review by shushen — APPROVED** [review]
Thanks @shawnmclean !

**Comment by shawnmclean** [comment]
This PR is blocked by `auth0@2.23.0` availability in webtask.

Awaiting roll out.

**Comment by shawnmclean** [comment]
Just giving an update here, we are currently rolling out the dependencies that this update requires. We have tested it and now awaiting the rollout which may take a day or two next week.

**Pull request merged**